### ISSUE TITLE
Decide on default content caching setting

### ISSUE THREAD

**Issue by smartass101** [getpelican/pelican#1689]
I think it's obvious that content caching is a useful setting and it does speed up things. However, since 3.4 there have been far too many instances when people encountered a problem that was due to caching being on by default, particulary when experimenting with different options at the beginning of their pelican usage.

I think we should decide whether to turn caching off by default in 3.6 and mention in docs that it can be turned on.

Rationale:
* Most new users will start by experimenting with settings and will enable caching once they arrive at a final shape of their website.
* Long-term users can just add enable caching if they want to continue to use it. Until they do, the original cache will still be there in their cache/ folder so they won't lose anything. However, it would be fair to log a message of some sort that caching is off.

Not sure what the "caching is off" warning should look like. maybe something like in settings.py
```Python
if not settings['LOAD_CONTENT_CACHE']:
    logger.info('Content cache not loaded, all content will be re-read')
if not settings['CACHE_CONTENT']:
    logger.info('Content caching is disabled, newly read content will be re-read next time')
```
These info messages would also make things clearer in general IMHO.

What do you think?
Once decided, I'll commit the changes to the settings, docs and logging.

**Comment by iKevinY** [comment]
I agree with making this change. The progression from experimenting with site options to learning how to enable the cache seems more logical than the current situation.

**Comment by avaris** [comment]
Given that I advocated this for 3.5, I'm all for it.

**Comment by ingwinlu** [comment]
go for it.

**Comment by smartass101** [comment]
What do you think @justinmayer?

**Comment by justinmayer** [comment]
I agree that disabled-by-default is probably best from a *principle of least astonishment* perspective.

I'm not sure that the warning is needed. If generation is slow, I presume folks will be incentivized to, ya know, read the docs and learn about caching. What are everyone's opinions on that?

**Comment by pxquim** [comment]
I'm not even certain that caching is such a big win, at least not always. For example, some plugins process articles into BeautifulSoup and back, and that this processing is not cached. I've only seen modest performance wins, which are offset by the occasional cache issue.

**Comment by smartass101** [comment]
@justinmayer, I was thinking of a warning mainly to give a  heads up and simplify debugging for people on large sites when suddenly the new version slows down.

@pxquim, it is mainly useful on larger sites. By default content is cached only at the reader level, but if plugins modify content extensively, the generator level might help a lot more.

**Comment by robclancy** [comment]
We have builds going past 10 minutes WITH caching, so yes it is needed. We have big sites.

But can you just decide on this and close it? Flip a coin.

**Comment by justinmayer** [comment]
I think we should change the default behavior (as discussed above) and put the information in the release notes. People who don't need caching shouldn't have to put up with constant warnings about things that aren't relevant to them, so I think we should skip the warning.

@smartass101: What do you think? Is this something you could address over the next week or so?

**Comment by justinmayer** [comment]
@smartass101: Would you be so kind as to make these changes this week? It would be nice to issue a release thereafter, as this is one of the few remaining blockers. Thanks, Ondrej!

**Comment by justinmayer** [comment]
@smartass101: Any word on this? Would you like someone else to implement the caching behavior change?

@ingwinlu: Since you are currently working on cache-related topics, is this something you'd care to address?

**Comment by smartass101** [comment]
Sorry for beying off the radar, have been very busy. I'm not sure wheher to go ahead with this or wait for @ingwinlu to overcome it. Ideally, I'd welcome @ingwinlu's solution, but I'm not sure whether it is realistic.

**Comment by ingwinlu** [comment]
#1745 still needs work, mainly
* don't save and compare shallow settings, use a hash of the current settings instead
* detect if a plugin got modified, (so a hash of the module or somekind) to detect those changes as well

before we can continue work on those effectively we should pool efforts on #1748 and fix the listed issues there, as the current generation is just a series of happy accidents and follows no structure.

That is why I opened up #1747 which includes the bugfixing portion of the earlier mentioned pr, which does not add additional functionality, (see the commit for more information). I am happy to add a second commit to that PR which addresses this issue via switching default cache off **but** current cache settings should be simplified or removed
* `AUTORELOAD_IGNORE_CACHE` beyond useless -> remove
*  merge `LOAD_CONTENT_CACHE` and `CACHE_CONTENT` into `CACHE_CONTENT`
* set `CACHE_CONTENT = FALSE` as default

There is no need to address documentation imo.

**Comment by justinmayer** [comment]
Addressed via #1747. Thanks, Winlu!
